Source/Diagnostics/ReducedDiags/LoadBalanceCosts.H

```diff
@@ -19,6 +23,19 @@ class LoadBalanceCosts : public ReducedDiags
 {
 public:
 
+    /** stores the host identifier (string) */
+    char* m_data_string_recvbuf = nullptr;
```

Comment:
you could store those as `std::vector< char >`, allocate it with `m_data_string_recvbuf.resize(MPI_MAX_PROCESSOR_NAME+1);` and and write into it via its `.data()` member to avoid the C-style memory handling.

Reply:
perfect, I tried (and failed valiantly) to do something like this earlier, but the way you outlined is what I was looking for.  the update now uses `Vector<char>` and we can get rid of the C-style memory management.